panther_power_control/src/power_board_node.py

```diff
@@ -128,45 +124,51 @@ def __init__(self, name: str) -> None:
         #   Timers
         # -------------------------------
 
-        # 2 Hz chearger publishing timer
-        self._charger_state_timer = rospy.Timer(rospy.Duration(0.5), self._publish_charger_state_cb)
-        # 10 Hz e-stop publishing timer
-        self._e_stop_timer = rospy.Timer(rospy.Duration(0.1), self._publish_e_stop_state_cb)
-        # 50 Hz of software PWM
+        # 2 Hz publish non asynch pin state
+        self._charger_state_timer = rospy.Timer(rospy.Duration(0.5), self._publish_pin_state_cb)
+        # 50 Hz of software PWM. Timer running at 100 Hz for raising and falling edges
         self._watchdog_timer = rospy.Timer(rospy.Duration(0.01), self._watchdog_cb)
         
         # -------------------------------
         #   GPIO callbacks
         # -------------------------------
     
+        # for fast e-stop detection
+        GPIO.add_event_detect(self._pins.E_STOP_RESET, GPIO.BOTH,
+                              callback=self._gpio_interrupt_cb, bouncetime=200)
+        
         GPIO.add_event_detect(self._pins.SHDN_INIT, GPIO.RISING,
-                              callback=self._soft_shutdow_cb, bouncetime=200)
-
+                              callback=self._gpio_interrupt_cb, bouncetime=200)
+        
         rospy.loginfo(f'[{rospy.get_name()}] Node started')
         
     def _cmd_vel_cb(self, *args) -> None:
         self._cmd_vel_msg_time = rospy.get_time()
-
-    def _soft_shutdow_cb(self, pin) -> None:
+        
+    def _gpio_interrupt_cb(self, pin: int) -> None:
         if pin == self._pins.SHDN_INIT:
             rospy.loginfo(f'[{rospy.get_name()}] Shutdown button pressed.')
-            self._power_button_pressed_pub.publish(Bool(True))
-
-    def _publish_charger_state_cb(self, *args) -> None:
+            self._publish_io_state('power_btn', True)
+            
+        if pin == self._pins.E_STOP_RESET:
+            self._e_stop_event()
+            
+    def _publish_pin_state_cb(self, *args) -> None:
         charger_state = self._read_pin(self._pins.CHRG_SENSE)
-        if self._charger_state_pub.impl.latch.data != charger_state:
-            msg = Bool(charger_state)
-            self._charger_state_pub.publish(msg)
+        self._publish_io_state('charger_connected', charger_state)
         
-    def _publish_e_stop_state_cb(self, *args) -> None:
-        msg = Bool(self._read_pin(self._pins.E_STOP_RESET))
-        self._e_stop_state_pub.publish(msg)
+        # to ensure correct e-stop state is published
+        self._e_stop_event()
+            
 
     def _watchdog_cb(self, *args) -> None:
         self._watchdog()
         
     def _aux_power_enable_cb(self, req: SetBoolRequest) -> SetBoolResponse:
-        return self._set_bool_srv_handle(req.data, self._pins.AUX_PW_EN, 'Aux power enable')
+        ret = self._set_bool_srv_handle(req.data, self._pins.AUX_PW_EN, 'Aux power enable')
+        if ret.success:
+            self._publish_io_state('aux_power', req.data)
+        return ret
```

Comment:
```suggestion
        res = self._set_bool_srv_handle(req.data, self._pins.AUX_PW_EN, 'Aux power enable')
        if res.success:
            self._publish_io_state('aux_power', req.data)
        return res
```
res as for response seems more intuitive